vault/resource_consul_secret_backend_test.go

```diff
@@ -24,83 +29,128 @@ func TestConsulSecretBackend(t *testing.T) {
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "max_lease_ttl_seconds", "0"),
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "address", "consul.domain.tld:8501"),
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "token", token),
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "scheme", "https"),
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "local", "false"),
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "ca_cert", "FAKE-CERT-MATERIAL"),
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "client_cert", "UPDATED-FAKE-CLIENT-CERT-MATERIAL"),
-					resource.TestCheckResourceAttr("vault_consul_secret_backend.test", "client_key", "UPDATED-FAKE-CLIENT-CERT-KEY-MATERIAL"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldPath, path),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldDescription, "test description"),
+					resource.TestCheckResourceAttr(resourceName, "default_lease_ttl_seconds", "0"),
+					resource.TestCheckResourceAttr(resourceName, "max_lease_ttl_seconds", "0"),
+					resource.TestCheckResourceAttr(resourceName, "address", "consul.domain.tld:8501"),
+					resource.TestCheckResourceAttr(resourceName, "token", token),
+					resource.TestCheckResourceAttr(resourceName, "scheme", "https"),
+					resource.TestCheckResourceAttr(resourceName, consts.FieldLocal, "false"),
+					resource.TestCheckResourceAttr(resourceName, "ca_cert", "FAKE-CERT-MATERIAL"),
+					resource.TestCheckResourceAttr(resourceName, "client_cert", "UPDATED-FAKE-CLIENT-CERT-MATERIAL"),
+					resource.TestCheckResourceAttr(resourceName, "client_key", "UPDATED-FAKE-CLIENT-CERT-KEY-MATERIAL"),
+				),
+			},
+			testutil.GetImportTestStep(resourceName, false, "token", "ca_cert", "client_cert", "client_key"),
+		},
+	})
+}
+
+func TestConsulSecretBackend_Bootstrap(t *testing.T) {
+	backend := acctest.RandomWithPrefix("tf-test-backend")
+	resourceName := "vault_consul_secret_backend.test"
+	resourceRoleName := "vault_consul_secret_backend_role.test"
+
+	if !testutil.CheckTestVaultVersion(t, "1.11") {
+		t.Skipf("test requires Vault 1.11 or newer")
+	}
+
+	cleanup, consulConfig := consulhelper.PrepareTestContainer(t, "1.12.3", false, false)
+	t.Cleanup(cleanup)
+	consulAddr := consulConfig.Address()
+
+	resource.Test(t, resource.TestCase{
+		Providers:    testProviders,
+		PreCheck:     func() { testutil.TestAccPreCheck(t) },
+		CheckDestroy: testAccConsulSecretBackendCheckDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testConsulSecretBackend_bootstrapConfig(backend, consulAddr),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr(resourceName, consts.FieldPath, backend),
+					resource.TestCheckResourceAttr(resourceName, "address", consulAddr),
+				),
+			},
```

Comment:
suggest testing import after each config test step:
```suggestion
			},
			testutil.GetImportTestStep(resourceName, false),